packages/nextcloud/packages/nextcloud_test_presets/lib/src/generate_presets.dart

```diff
@@ -63,31 +63,32 @@ Future<void> generatePresets() async {
   }
   serverPresetsDir.createSync();
 
-  for (final serverVersion in serverVersions) {
-    final buffer = StringBuffer()..writeln('SERVER_VERSION=$serverVersion');
+  for (final serverRelease in serverReleases) {
+    final buffer = StringBuffer()..writeln('SERVER_VERSION=${serverRelease.version}@${serverRelease.digest}');
 
     for (final app in apps) {
-      final release = app.findLatestCompatibleRelease(serverVersion) ??
-          app.findLatestCompatibleRelease(serverVersion, allowUnstable: true) ??
+      final release = app.findLatestCompatibleRelease(serverRelease) ??
+          app.findLatestCompatibleRelease(serverRelease, allowUnstable: true) ??
           app.findLatestRelease();
       buffer.writeln('${app.id.toUpperCase()}_URL=${release.url}');
     }
 
-    File('${serverPresetsDir.path}/${serverVersion.major}.${serverVersion.minor}').writeAsStringSync(buffer.toString());
+    File('${serverPresetsDir.path}/${serverRelease.version.major}.${serverRelease.version.minor}')
```

Comment:
The `"$major.$minor"` thing should also be moved into the class (maybe a getter called preset).